aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import xml.etree.ElementTree as ET
+from copy import copy
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from dask import delayed
+from fsspec.spec import AbstractFileSystem
+
+from .. import constants, exceptions, types
+from ..dimensions import (
+    DEFAULT_CHUNK_DIMS,
+    DEFAULT_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES_AND_SAMPLES,
+    DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES,
+    REQUIRED_CHUNK_DIMS,
+    DimensionNames,
+)
+from ..utils import io_utils
+from .reader import Reader
+
+try:
+    from _aicspylibczi import BBox
+    from aicspylibczi import CziFile
+
+except ImportError:
+    raise ImportError(
+        "aicspylibczi is required for this reader. "
+        "Install with `pip install aicsimageio[czi]`"
+    )
+
+DEFAULT_CZI_CHUNK_DIMS = [
+    dim if dim != DimensionNames.Samples else "A" for dim in DEFAULT_CHUNK_DIMS
+]
+REQUIRED_CZI_CHUNK_DIMS = [
+    dim if dim != DimensionNames.Samples else "A" for dim in REQUIRED_CHUNK_DIMS
+]
+DEFAULT_CZI_DIMENSION_ORDER_LIST = [
+    dim if dim != DimensionNames.Samples else "A"
+    for dim in DEFAULT_DIMENSION_ORDER_LIST_WITH_SAMPLES
+]
+DEFAULT_CZI_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES = [
+    dim if dim != DimensionNames.Samples else "A"
+    for dim in DEFAULT_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES_AND_SAMPLES
+]
+
+
+###############################################################################
+
+
+class CziReader(Reader):
+    """
+    Wraps the aicspylibczi API to provide the same aicsimageio Reader API but for
+    volumetric Zeiss CZI images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    chunk_by_dims: Union[str, List[str]]
+        Which dimensions to create chunks for.
+        Default: DEFAULT_CHUNK_DIMS
+        Note: Dimensions.SpatialY, Dimensions.SpatialX, and DimensionNames.Samples,
+        will always be added to the list if not present during dask array
+        construction.
+
+    Notes
+    -----
+    To use this reader, install with: `pip install aicsimageio[czi]`.
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            with fs.open(path) as open_resource:
+                CziFile(open_resource)
+                return True
+
+        except RuntimeError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        chunk_by_dims: Union[str, List[str]] = DEFAULT_CHUNK_DIMS,
+    ):
+        # doesn't do anything in this case but here for completeness
+        super(CziReader, self).__init__(image=image)
```

Comment:
In this case, this does nothing but I'm kinda sure this is proper practice for any class with a base class.
Yes/No?

Reply:
I think the standard behavior is that if you are inheriting from something that has an __init__ to do it. Since you can actually initialize an ABC I dont think this is standard.

So remove.